Assets/Scripts/UI/DeckDisplay.cs

```diff
@@ -78,35 +78,24 @@ public void LoadAllSupportCards(Dictionary<ulong, int> availableCardIds, Diction
         }
     }
 
-    private void AddCard(ulong cardId, int count, bool isSelected, bool isSupportCard)
+    private void AddCard(ulong cardId, int count, bool isSelected, bool supportCard)
     {
         if (cardId == 0)
         {
             // if this shows up there could be a bug in the contract
             Debug.LogWarning("Invalid card");
             return;
         }
-        var metadata = PepemonFactoryCardCache.GetMetadata(cardId);
+        var cardIsSupportCard = PepemonFactoryCardCache.GetMetadata(cardId)?.isSupportCard ?? false;
 
-        var cardAttribute = new CardAttribute 
-        { 
-            value = isSupportCard ? "Pepemon Support" : "Pepemon Battle",
-            trait_type = "Set" 
-        };
-
-        // skip battlecards if isSupportCard=true and skip supportcards if isSupportCard=false
-        if (!metadata?.attributes.Contains(cardAttribute) ?? false)
+        // skip battlecards if supportCard=true and skip supportcards if supportCard=false
+        if (cardIsSupportCard ^ supportCard)
```

Comment:
!= might be easier to follow here
```suggestion
        if (cardIsSupportCard != supportCard)
```